### PR TITLE
docs: add readme to vtl resolvers directory

### LINKED ISSUE #6325: Add README to VTL resolvers directory
This is hopefully a simple DX idea.

I didn't know you could over-write generated resolvers as described here: https://docs.amplify.aws/cli/graphql-transformer/resolvers

I thought the ``/amplify/backend/api/<MyApp>/resolvers`` folder was just where all resolvers lived. I didn't know that they actually all get generated into ``/amplify/backend/api/<MyApp>/build/resolvers`` and any that appear in ``/amplify/backend/api/<MyApp>/resolvers`` are over-rides or custom resolvers that will replace the generated one. 

This is actually really clever but the convention was hard for me to find, I had to have my build broken for 3 weeks to learn this bit (See https://github.com/aws-amplify/amplify-cli/issues/6315 to follow along at home).

Might I suggest that folders with this over-rides could be renamed as ``customResolvers`` rather than ``resolvers`` or something similar? I know that on the face of it this is a minor change but it will have knock on effects and it could be the genie is out of the bottle and its too late to change it. But that to me would have helped with my mental model of what that folder actually is for.

If we went a step forward could it be worth dropping a readme into these folders explaining what they are, when you might use them and links to the online docs? I only found this gem of a page today (https://docs.amplify.aws/cli/reference/files) and if nothing else it would be nice to add something in there about where to put custom code that over-rides the generated stuff.

### PR DESCRIPTION
**Pull request by rajrajhans**
*Issue #, if available:* Closes #6325

*Description of changes:*
Adds a README to `amplify/backend/api/<MyApp>/resolvers` directory describing the override process for resolvers. More details in the linked issue.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by edwardfoyle — COMMENTED** [review]
Hi @rajrajhans thanks for the contribution! I think it would be better to add the README as a separate file in the resources directory of `amplify-category-api` and then copy the file into the project destination rather than defining the readme as a string. The file could go under a path like `amplify-category-api/resources/awscloudformation/resolver-readme/README.md`.

Also, it looks like a lint check is failing, so double check that.

cc @renebrandel to comment on the wording of the README

**Comment by renebrandel** [comment]
Wording looks good to me.

**Comment by rajrajhans** [comment]
@edwardfoyle Thank you for the review. I have made the required changes, and fixed the lint issues as well.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts [hunk 1]
```diff
@@ -59,6 +59,7 @@ class CfnApiArtifactHandler implements ApiArtifactHandler {
     const stacksDirectoryPath = path.join(resourceDir, stacksDirName);
     if (!fs.existsSync(stacksDirectoryPath)) {
       fs.mkdirSync(stacksDirectoryPath);
+      fs.copyFileSync(path.join(rootAssetDir, 'resolver-readme', 'README.md'), `${resolverDirectoryPath}/README.md`);
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts:62]: nit use `path.join()` for the destination path as well

Comment (rajrajhans) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts]: Done

**Review by edwardfoyle — APPROVED** [review]
LGTM! Thanks for addressing the comments!

**Pull request merged**